resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -268,27 +268,19 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
         .getOrElse {
           throw new SparkException(s"Executor Spark home `$EXECUTOR_HOME` is not set!")
         }
-      val runScript = new File(executorSparkHome, "./bin/spark-class").getPath
-      command.setValue(
-        "%s \"%s\" org.apache.spark.executor.CoarseGrainedExecutorBackend"
-          .format(prefixEnv, runScript) +
-        s" --driver-url $driverURL" +
-        s" --executor-id $taskId" +
-        s" --hostname ${executorHostname(offer)}" +
-        s" --cores $numCores" +
-        s" --app-id $appId")
+      val executable = new File(executorSparkHome, "./bin/spark-class").getPath
+      val runScript = s"$prefixEnv $executable " +
```

Comment:
Do you need to quote `$executable`? also the second string interpolation below here isn't needed.

Reply:
The resulting run script takes the following form if `spark.executor.extraLibraryPath=path` is provided:
```
LD_LIBRARY_PATH=path:$LD_LIBRARY_PATH org.apache.spark.executor.CoarseGrainedExecutorBackend
```
so there's no reason for quating the actual executable. Thanks for pointing out to unneeded interpolation.

Reply:
I wonder if it's quoted to account for, say, paths with spaces? That's the reason I've seen this done elsewhere. I'd keep it if it does no harm.

Reply:
Good point, thanks @srowen. Added the quotes back.